index.js

```diff
@@ -18,149 +10,197 @@ var mkdirp = require('mkdirp');
  * function is not passed.
  *
  * ```js
- * var writeFile = require('write');
- * writeFile('foo.txt', 'This is content...', function(err) {
- *   if (err) console.log(err);
- * });
+ * const write = require('write');
+ *
+ * // async/await
+ * (async () => {
+ *   await write('foo.txt', 'This is content...');
+ * })();
  *
  * // promise
- * writeFile('foo.txt', 'This is content...')
- *   .then(function() {
+ * write('foo.txt', 'This is content...')
+ *   .then(() => {
  *     // do stuff
  *   });
+ *
+ * // callback
+ * write('foo.txt', 'This is content...', err => {
+ *   // do stuff with err
+ * });
  * ```
- * @name writeFile
- * @param {string|Buffer|integer} `filepath` filepath or file descriptor.
- * @param {string|Buffer|Uint8Array} `data` String to write to disk.
- * @param {object} `options` Options to pass to [fs.writeFile][fs]{#fs_fs_writefile_file_data_options_callback} and/or [mkdirp][]
+ * @name write
+ * @param {String} `filepath` file path.
+ * @param {String|Buffer|Uint8Array} `data` Data to write.
+ * @param {Object} `options` Options to pass to [fs.writeFile][writefile]
  * @param {Function} `callback` (optional) If no callback is provided, a promise is returned.
  * @api public
  */
```

Comment:
These comments could use some information on the return value. I noticed there's a `result` object with `path` and `data` returned. I think this is a good idea, especially when `increment` is used.

Reply:
done